example/brunel/brunel_miniapp.cpp

```diff
@@ -240,37 +235,31 @@ int main(int argc, char** argv) {
 
         brunel_recipe recipe(nexc, ninh, next, in_degree_prop, w, d, rel_inh_strength, poiss_lambda, seed);
 
-        auto register_exporter = [] (const io::cl_options& options) {
-            return std::make_unique<io::exporter_spike_file>
-                       (options.file_name, options.output_path,
-                        options.file_extension, options.over_write);
-        };
+        partition_hint_map hints;
+        hints[cell_kind::lif_neuron].cpu_group_size = group_size;
+        auto decomp = partition_load_balance(recipe, nd, &context, hints);
 
-        auto decomp = decompose(recipe, group_size, &context);
         simulation sim(recipe, decomp, &context);
 
         // Initialize the spike exporting interface
-        std::unique_ptr<io::exporter_spike_file> file_exporter;
+        std::fstream spike_out;
         if (options.spike_file_output) {
-            if (options.single_file_per_rank) {
-                file_exporter = register_exporter(options);
+            using std::ios_base;
 
-                sim.set_local_spike_callback(
-                    [&](const std::vector<spike>& spikes) {
-                        file_exporter->output(spikes);
-                    }
-                );
+            auto rank = context.id();
+            aux::path p = options.output_path;
+            p /= aux::strsub("%_%.%", options.file_name, rank, options.file_extension);
+
+            if (options.single_file_per_rank) {
```

Comment:
How about a single file for the whole simulation? For a demo, I don't see any point making this an option?
I suppose it was this way in the past to test or justify the existence of this option in the spike_exporter.

Reply:
The original code (which was copied from the original miniapp) supported either having each rank write their own spikes, or instead having rank 0 write all of them (see the `set_global_spike_callback` a few lines below). We can certainly nuke this, but here I'm just preserving the existing behaviour.